be/src/olap/storage_engine.cpp

```diff
@@ -1022,6 +1027,39 @@ void StorageEngine::_clean_unused_pending_publish_info() {
     }
 }
 
+void StorageEngine::_clean_unused_partial_update_info() {
+    std::vector<std::pair<int64_t, RowsetId>> remove_infos;
+    auto unused_partial_update_info_collector = [this, &remove_infos](
+                                                        int64_t tablet_id, RowsetId rowset_id,
+                                                        std::string_view value) -> bool {
+        TabletSharedPtr tablet = _tablet_manager->get_tablet(tablet_id);
+        if (tablet == nullptr) {
+            remove_infos.emplace_back(tablet_id, rowset_id);
+            return true;
+        }
+        RowsetSharedPtr rowset = tablet->get_rowset(rowset_id);
+        if (rowset == nullptr) {
```

Comment:
should check txn id, rather than rowset
Whether the rowset is in the tablet or not, there is no way to confirm that this partial update info is useless, only use the txn id can confirm that. And you can't find a rowset in the tablet until it's published.